src/transaction/transaction.rs

```diff
@@ -664,6 +670,9 @@ impl<PdC: PdClient> Transaction<PdC> {
             .plan();
         let pairs = plan.execute().await;
 
+        // primary key will be set here if needed
+        self.buffer.get_primary_key_or(&first_key).await;
```

Comment:
This seems like an odd API - I think it should be something like `primary_key_or` and not return the existing primary key. Or we could keep the call earlier and use `get_primary_key_or_set` and return an empty key if it was not previously set.

Reply:
How about just renaming it to `get_primary_key_or_insert`? Similar to `Option::get_or_insert`.

Reply:
My point is not so much the name, but that this is the only place it is used and the caller throws away the returned value.

Reply:
Oh that makes sense.